homeassistant/components/scrape/sensor.py

```diff
@@ -144,6 +147,45 @@ async def async_setup_platform(
     async_add_entities(entities)
 
 
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Scrape sensor entry."""
+    entities: list = []
+
+    coordinator: ScrapeCoordinator = hass.data[DOMAIN][entry.entry_id]
+    config = dict(entry.options)
+    for sensor in config["sensors"]:
+        sensor_config: ConfigType = vol.Schema(
+            TEMPLATE_SENSOR_BASE_SCHEMA.schema, extra=vol.ALLOW_EXTRA
+        )(sensor)
+
+        name: str = sensor_config[CONF_NAME]
+        select: str | None = sensor_config.get(CONF_SELECT)
+        attr: str | None = sensor_config.get(CONF_ATTRIBUTE)
+        index: int = int(sensor_config[CONF_INDEX])
```

Comment:
```suggestion
        index: int = sensor_config[CONF_INDEX]
```

Reply:
This comes in as a float. Don't we want to force it into an int?
Not sure how this would be handled.

Reply:
We should force it into an int inside `validate_sensor_setup` I think